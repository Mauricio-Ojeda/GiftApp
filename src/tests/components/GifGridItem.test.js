import React from 'react';
import { shallow } from 'enzyme'
import GifGridItem from '../../components/GifGridItem'


describe('pruebas en GifGridItem', () => {
    
    const title = 'Soy un titulo';
    const url = 'https://soyunaurl.localhost.jpg';
    const wrapper = shallow( <GifGridItem
                                    title= { title }
                                    url = { url }
                             /> );
    
    test('debe mostrar el componente en un snapshot', () => {
        

        expect( wrapper ).toMatchSnapshot();

        
    });

    test('debe tener un parrafo con el title', () => {
        
        const testTitle = wrapper.find('p');
        expect( testTitle.text().trim() ).toBe( title );

    });
    
    test('debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        
        
        
    })
    

})
