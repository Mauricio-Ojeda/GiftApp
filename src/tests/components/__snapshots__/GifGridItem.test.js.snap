// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem debe mostrar el componente en un snapshot 1`] = `
<div
  className="card animate__lightSpeedInRight animate__fadeIn"
>
  <img
    alt="Soy un titulo"
    src="https://soyunaurl.localhost.jpg"
  />
  <p>
    Soy un titulo
  </p>
</div>
`;
